### PR TITLE
RFC 2070 part 1: PanicInfo and Location API changes

### PR DESCRIPTION
**Pull request by SimonSapin**
This implements part of https://rust-lang.github.io/rfcs/2070-panic-implementation.html
Tracking issue: https://github.com/rust-lang/rust/issues/44489

* Move `std::panic::PanicInfo` and `std::panic::Location` to a new `core::panic` module. The two types and the `std` module were already `#[stable]` and stay that way, the new `core` module is `#[unstable]`.
* Add a new `PanicInfo::message(&self) -> Option<&fmt::Arguments>` method, which is `#[unstable]`.
* Implement `Display` for `PanicInfo` and `Location`

### CODE REVIEW

File: src/libcore/panic.rs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    ///     if let Some(location) = panic_info.location() {
+    ///         println!("panic occurred at column {}", location.column());
+    ///     } else {
+    ///         println!("panic occurred but can't get location information...");
+    ///     }
+    /// }));
+    ///
+    /// panic!("Normal panic");
+    /// ```
+    #[stable(feature = "panic_col", since = "1.25")]
```
Comment (kennytm) [inline src/libcore/panic.rs]: Let's correct this to `1.25.0`

**Comment by alexcrichton** [comment]
@rfcbot fcp merge

Looks great to me, thanks @SimonSapin!

File: src/libcore/panic.rs (hunk 1 above)
Comment (SimonSapin) [reply src/libcore/panic.rs]: Done, and other similar occurrences.

File: src/libcore/panic.rs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            reason = "newly available in libcore",
+            issue = "44489")]
+
+use any::Any;
+use fmt;
+
+/// A struct providing information about a panic.
+///
+/// `PanicInfo` structure is passed to a panic hook set by the [`set_hook`]
+/// function.
```
Comment (whitequark) [inline src/libcore/panic.rs]: I think the documentation should mention the no-std case.

**Comment by shepmaster** [comment]
Ping for ticky boxes @BurntSushi, @Kimundi!

**Comment by SimonSapin** [comment]
> No output has been received in the last 30m0s, this potentially indicates a stalled build or something wrong with the build itself.

**Pull request merged**